From ae608ea2735bd331ec7dcf9d89928c38f0e0c981 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Mon, 6 Mar 2017 18:59:22 +0100
Subject: [PATCH] Fix static compilation of json2cbor

json2cbor depends on cjson so tools/json2cbor/json2cbor.o must be
before -lcjson

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream commit: https://github.com/01org/tinycbor/commit/ae608ea2735bd331ec7dcf9d89928c38f0e0c981]
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 97b292c..a8cb57c 100644
--- a/Makefile
+++ b/Makefile
@@ -115,7 +115,7 @@ bin/cbordump: $(CBORDUMP_SOURCES:.c=.o) lib/libtinycbor.a
 
 bin/json2cbor: $(JSON2CBOR_SOURCES:.c=.o) lib/libtinycbor.a
 	@$(MKDIR) -p bin
-	$(CC) -o $@ $(LDFLAGS) $(LDFLAGS_CJSON) $^ $(LDLIBS) -lm
+	$(CC) -o $@ $(LDFLAGS) $^ $(LDFLAGS_CJSON) $(LDLIBS) -lm
 
 tinycbor.pc: tinycbor.pc.in
 	$(SED) > $@ < $< \
